File: 1984.Minimum-Difference-Between-Highest-and-Lowest-of-K-Scores.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function (nums, k) {
  if (k === 1) return 0;

  nums.sort((a, b) => a - b);
  let result = Infinity;

  for (let i = 0; i <= nums.length - k; i++) {
    const j = i + k - 1;

    result = Math.min(result, nums[j] - nums[i]);
  }

  return result;
};

File: python/1984.Minimum-Difference-Between-Highest-and-Lowest-of-K-Scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if k == 1:
            return 0

        nums.sort()
        result = float("inf")

        for i in range(len(nums) - k + 1):
            j = i + k -1
            result = min(result, nums[j] - nums[i])

        return result